File: client/android/app/src/main/java/org/cna/keyple/demo/sale/android/nfc/slave/di/DataModule.java
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 ********************************************************************************/
package org.cna.keyple.demo.sale.android.nfc.slave.di;

import org.cna.keyple.demo.sale.android.nfc.slave.Application;
import org.cna.keyple.demo.sale.android.nfc.slave.di.scopes.AppScoped;
import android.content.Context;
import android.content.SharedPreferences;
import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    private String PREF_NAME = "Keyple-prefs";

    @Provides
    @AppScoped
    SharedPreferences getSharedPreferences(Application app) {
        return app.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }
}

File: client/android/app/src/main/kotlin/org/eclipse/keyple/demo/remote/di/DataModule.kt
/*
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information
 * regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */


package org.eclipse.keyple.demo.remote.di

import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import org.eclipse.keyple.demo.remote.Application
import org.eclipse.keyple.demo.remote.di.scopes.AppScoped

@Module
class DataModule {
    private val PREF_NAME = "Keyple-prefs"
    @Provides
    @AppScoped
    fun getSharedPreferences(app: Application): SharedPreferences {
        return app.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }
}